### PR TITLE
Add GUID::to_u128

### LINKED ISSUE #1755: Feature request: GUID to u128 conversion function
### Motivation

Well, there is `GUID::from_u128`, so I think, there should be `GUID::to_u128`. Then we could use u128 for public interface in crates and convert it to GUID.

### Drawbacks

A bit slower compilation time?

### Rationale and alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GamePad64**
Now GUID can round-trip to and from u128

Relates to #1306
Closes #1755

### CODE REVIEW

**Comment by kennykerr** [comment]
Would a `From` impl be more appropriate? `from_u128` exists because `From` is not `const` and needs to be for the `GUID` constants in the `windows` crate.

**Comment by GamePad64** [comment]
Hmm, implementing From looks good for me. But having a separate `const fn` is nice, because compiler can optimize it out and it may matter in some cases.

**Comment by kennykerr** [comment]
I think `From` is more idiomatic - how about we drop the explicit `from_u128` function for now - we can always add a `const` function down the road if that becomes necessary. Then we just need a basic test in `crates/tests/core/tests/guid.rs`.

**Comment by kennykerr** [comment]
Oh, I see you have one already. I guess there is some symmetry to it. Let's see what others think.

File: crates/libs/windows/src/core/guid.rs [hunk 1]
```diff
@@ -98,6 +103,24 @@ impl core::convert::From<&str> for GUID {
...
+    }
+}
+
+impl core::convert::From<GUID> for u128 {
+    fn from(value: GUID) -> Self {
+        value.to_u128()
+    }
+}
+
+impl core::convert::From<&GUID> for u128 {
```
Comment (rylev) [inline crates/libs/windows/src/core/guid.rs:118]: We don't impl `From<&u128> for GUID` so I'm not sure we need to do the reverse. `GUID` is also `Copy` so it's trivial to go from a `&GUID` to `GUID`.

**Review by rylev — COMMENTED** [review]
I like the symmetry. Looks good to mem just with one comment

File: crates/libs/windows/src/core/guid.rs (hunk 1 above)
Comment (GamePad64) [reply crates/libs/windows/src/core/guid.rs]: Ok, removed From<&GUID>

**Review by rylev — APPROVED** [review]

**Comment by rylev** [comment]
Thanks for the contribution! Looks great 😊

**Pull request merged**